core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2482,6 +2482,50 @@ wasm_runtime_get_custom_data(WASMModuleInstanceCommon *module_inst_comm)
     return module_inst->custom_data;
 }
 
+#if WASM_CONFIGUABLE_BOUNDS_CHECKS != 0
+void
+wasm_runtime_set_bounds_checks(WASMModuleInstanceCommon *module_inst,
+                               bool enable)
+{
+#if WASM_ENABLE_INTERP != 0
+    if (module_inst->module_type == Wasm_Module_Bytecode) {
+        ((WASMModuleInstanceExtra *)((WASMModuleInstance *)module_inst)->e)
+            ->disable_bounds_checks = enable ? false : true;
+    }
+#endif
+
+#if WASM_ENABLE_AOT != 0
+    if (module_inst->module_type == Wasm_Module_AoT) {
+        ((AOTModuleInstanceExtra *)((AOTModuleInstance *)module_inst)->e)
+            ->disable_bounds_checks = enable ? false : true;
+    }
+#endif
+}
+#endif
+
+bool
+wasm_runtime_is_bounds_checks_enabled(WASMModuleInstanceCommon *module_inst)
+{
+#if WASM_CONFIGUABLE_BOUNDS_CHECKS != 0
+#if WASM_ENABLE_INTERP != 0
+    if (module_inst->module_type == Wasm_Module_Bytecode) {
+        return !((WASMModuleInstanceExtra *)((WASMModuleInstance *)module_inst)
+                     ->e)
+                    ->disable_bounds_checks;
```

Comment:
Same as above, when OS_ENABLE_HW_BOUND_CHECK is defined and hw bound check is enabled, shall we always return false here?

Reply:
@wenyongh Emm, In this patch there is a extra behavior that map the app addr 0 to host addr NULL, this is a non standard option, so can't pass the spec test with OS_ENABLE_HW_BOUND_CHECK.

Should we add a extra option to control this behavior?

Reply:
Maybe we should do the mapping in wrapper, I'll remove it later.

Reply:
OK, so don't add control for OS_ENABLE_HW_BOUND_CHECK in this function.